README.md

```diff
@@ -3,3 +3,104 @@
 Knative is a Kubernetes-based platform to deploy and manage modern serverless workloads.
 
 For more details of what is possible with Knative, see https://knative.dev/
+
+## Usage
+
+### Pre-requisites
+
+Microk8s 1.21/stable
+>NOTE: These instructions assume you have run `microk8s enable dns storage rbac metallb:"10.64.140.43-10.64.140.49,192.168.0.105-192.168.0.111"`
+
+### Add a model and set variables
+
+```
+MODEL_NAME="knative"
+DEFAULT_GATEWAY="knative-gateway"
+juju add-model ${MODEL_NAME}
+```
+
+### Deploy dependencies
+
+Knative (in particular `knative-serving`) requires `istio-operators` to be deployed in the model. To correctly configure them, you can:
+
+```
+juju deploy istio-pilot --config default-gateway=${DEFAULT_GATEWAY} --trust
+juju deploy istio-gateway istio-ingressgateway --config kind="ingress" --trust
+juju relate istio-pilot istio-ingressgateway
+```
+
+### `knative-operator`
+
+The `knative-operator` is responsible for installing, configuring, and managing the `serving` and `eventing` components. Despite your final application, you must deploy the charmed `knative-operator` before any other Knative component.
+
+```
+juju deploy knative-operator --trust
+```
+
+### `knative-serving`
+
+`knative-serving` provides components that enable rapid deployment of serverless containers, autoscaling (including down to zero), support for multiple layers, and revision tracking. Charmed `knative-serving` can be deployed as follows:
+
+```
+juju deploy knative-serving --config namespace="knative-serving" --config istio.gateway.namespace=${MODEL_NAME} --trust
+```
```

Comment:
Can you add explanations for these?  Something like below.  Also, I'd put the `istio.gateway.name`, even though it happens to work with the default.  Good chance someone deploying this wont actually use the `knative-gateway` name applied above, which then means they need to know about this config.


Sorry, gh doesn't like "suggestions" that include backticks.  Below is a suggested update to the highlighted section:

```
juju deploy knative-serving --config namespace="knative-serving" --config istio.gateway.namespace=${MODEL_NAME} --trust
```

where:
* `namespace`: The namespace `knative-serving` will be deployed into (it cannot be deployed into the same namespace as `knative-operator` or `knative-eventing`
* istio.gateway.namespace: The namespace the Istio gateway is deployed to (generally, the model that Istio is deployed to)
* istio.gateway.name: The name of the Istio gateway

Reply:
I just changed the `namespace` definition a bit, we have to specify it's the resources not the charm this namespace has effect on.